### PR TITLE
Drs bg service

### PR DESCRIPTION
**Pull request by josh-dadswell-hackney**
Add/Update appointments from DRS when a message is received

### CODE REVIEW

File: RepairsApi.Tests/E2ETests/Repairs/DRSBackgroundServiceE2ETests.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            startTime = startTime.AddTicks(-(startTime.Ticks % TimeSpan.TicksPerSecond));
+            var endTime = startTime.AddHours(5);
+
+            var result = await CreateWorkOrder(wo => wo.AssignedToPrimary.Organization.Reference.First().ID = TestDataSeeder.DRSContractor);
+
+            var response = await CreateAppointment(result, startTime, endTime);
+
+            var appointment = GetAppointmentFromDB(result.Id);
+
+            response.Should().NotBeNull();
```
Comment (Asura5) [inline RepairsApi.Tests/E2ETests/Repairs/DRSBackgroundServiceE2ETests.cs:44]: Do we have a rule about multiple asserts per test here?

File: RepairsApi.Tests/E2ETests/Repairs/WorkOrderHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public static class WorkOrderHelpers
+    {
+        public static Generator<T> CreateWorkOrderGenerator<T>()
+        {
+            var gen = new Generator<T>()
+                .AddWorkOrderGenerators()
+                .AddValue(new List<double>
+                {
+                    0
```
Comment (Asura5) [inline RepairsApi.Tests/E2ETests/Repairs/WorkOrderHelpers.cs:15]: If this is part of a testing suite, how about a randomly generated value?

File: RepairsApi.Tests/E2ETests/Repairs/DRSBackgroundServiceE2ETests.cs (hunk 1 above)
Comment (josh-dadswell-hackney) [reply RepairsApi.Tests/E2ETests/Repairs/DRSBackgroundServiceE2ETests.cs:44]: Generally no, I aggree 1 per unit test but I am normally more flexible with integration tests

File: RepairsApi/V2/Services/DRS/BackgroundService/DrsBackgroundServiceLogger.cs [hunk 3]
```diff
@@ -1,29 +1,60 @@
...
+using System.Net.Http;
+using System.Text;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.Logging;
+using Microsoft.IO;
 
 namespace RepairsApi.V2.Generated.DRS.BackgroundService
 {
+#pragma warning disable // Disable all warnings
```
Comment (Asura5) [inline RepairsApi/V2/Services/DRS/BackgroundService/DrsBackgroundServiceLogger.cs:12]: Can you be more specific with this pragma?

Comment (josh-dadswell-hackney) [reply RepairsApi/V2/Services/DRS/BackgroundService/DrsBackgroundServiceLogger.cs:12]: Oops yes I can

File: RepairsApi.Tests/E2ETests/Repairs/DRSBackgroundServiceE2ETests.cs (hunk 1 above)
Comment (Asura5) [reply RepairsApi.Tests/E2ETests/Repairs/DRSBackgroundServiceE2ETests.cs:44]: Yeah, that's fine, just checking

**Pull request merged**